### PR TITLE
ENH: stats: add `friston` and `nichols` methods to `combine_pvalues`

### LINKED ISSUE #15481: add extra methods to scipy.stats.combine_pvalues function
I would propose to also add the "Friston" and "Nichols" methods (I am not sure if these are the official names for these methods) for computing a combined p-value. See: https://stats.stackexchange.com/a/160804/94255. 

The statistic for both methods is `S = np.max(pvalues)`, and the combined p-value is `S**len(pvalues)` for the "Friston" method and `S` for the "Nichols" method.

### PR DESCRIPTION
**Pull request by fkloosterman**
#### Reference issue
Closes #15481

#### What does this implement/fix?
This commit adds two new methods to combine p-values derived from
multiple independent tests of the same hypothesis. Citations of
literature have been added to the doc string and the methods are
covered by new tests. See #15481.

### CODE REVIEW

**Comment by josef-pkt** [comment]
wrong title friston not fisher

File: scipy/stats/_stats_py.py [hunk 1]
```diff
@@ -8242,11 +8252,18 @@ def combine_pvalues(pvalues, method='fisher', weights=None):
         Zi = distributions.norm.isf(pvalues)
         statistic = np.dot(weights, Zi) / np.linalg.norm(weights)
         pval = distributions.norm.sf(statistic)
+    elif method == 'friston':
+        statistic = np.max(pvalues)
+        pval = statistic**len(pvalues)
+    elif method == 'nichols':
+        statistic = np.max(pvalues)
+        pval = statistic
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:8260]: @fkloosterman If you can point to the location within the references I can use to verify these, I might be able to review this PR.
 
Also, it seems that we should be able to interpret these results with a statement like "Under the null hypothesis, `pval` is the probability of achieving a value of the test statistic at least as extreme as `statistic`. What are the null hypotheses according to these methods? Perhaps another way of getting at this question is: How did you know what to return for `statistic`?

Comment (mdhaber) [reply scipy/stats/_stats_py.py:8260]: NM, looks like there is some discussion in gh-15481. Probably should come to a consensus there, and those involved can review this.

**Comment by tupui** [comment]
See https://github.com/scipy/scipy/issues/15481#issuecomment-1046705292

**Pull request closed without merging**